### PR TITLE
feat: add function to mask address

### PR DESCRIPTION
**Pull request by piavgh**

### CODE REVIEW

**Review by hieuvo — APPROVED** [review]

File: account/mask.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+func MaskAddress(address string, numVisibleChars int, maskedCharacter rune) (string, error) {
+	if !IsValidAddress(address) {
+		return "", ErrInvalidAddress
+	}
+
+	if numVisibleChars%2 != 0 {
+		return "", ErrInvalidNumVisibleChars
+	}
+
+	prefixLength := len("0x") + numVisibleChars/2 // The number of characters to keep at the start (0x + first some hex digits)
```
Comment (hieuvo) [inline account/mask.go:28]: minor, `numVisibleChars/2` vs `numVisibleChars / 2` 🤔

Comment (piavgh) [reply account/mask.go:28]: hmm, sometimes go formating is weird https://go.screenpal.com/watch/cZifeCV8zdQ

File: account/mask.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+// MaskAddress masks the middle part of an Ethereum address with a number of masked characters and returns the masked address.
+// The number of visible characters must be even.
+// The number of masked characters must be greater than 0.
+// The masked character can be any character.
+// @param address: The Ethereum address to mask
+// @param numVisibleChars: The number of visible characters to keep at the start and end of the address
+// @param maskedCharacter: The character to use for masking the middle part of the address
+// @return The masked address and error if any
+func MaskAddress(address string, numVisibleChars int, maskedCharacter rune) (string, error) {
```
Comment (hieuvo) [inline account/mask.go:19]: with negative numVisibleChars, this method will be panic

Comment (piavgh) [reply account/mask.go:19]: thanks, good catch

Comment (piavgh) [reply account/mask.go:19]: I pushed a commit to fix this

**Review by hieuvo — APPROVED** [review]

**Pull request merged**